include/libtorrent/aux_/session_impl.hpp

```diff
@@ -167,6 +170,8 @@ namespace libtorrent
 		{
 			// the size of each allocation that is chained in the send buffer
 			enum { send_buffer_size_impl = 128 };
+			// maximum length of query names which can be registered by extensions
+			enum { max_dht_query_length = 15 };
```

Comment:
since the member after the array will most likely be aligned to 8 bytes (pointer) and at least 4 bytes (for 32 bit systems). I would imagine this length could be 31 without any penalty.